lnwallet/channel.go

```diff
@@ -811,6 +812,24 @@ func (lc *LightningChannel) diskHtlcToPayDesc(feeRate chainfee.SatPerKWeight,
 		}
 	}
 
+	// Reconstruct the proper output index depending on whose commitment we
+	// are generateing and whether or not the HTLC is dust according to the
+	// other party's policy.
+	var (
+		localOutputIndex  int32
+		remoteOutputIndex int32
+	)
+	switch {
+	case isLocal && isDustLocal:
+		localOutputIndex = -1
+	case !isLocal && isDustRemote:
+		remoteOutputIndex = -1
```

Comment:
Do we need to explicitly set this, won't `htlc.OutputIndex` already be written as `-1` in the DB?

Reply:
good point! will update